eth/tracers/native/erc7562.go

```diff
@@ -0,0 +1,550 @@
+	Error        string          `json:"error,omitempty" rlp:"optional"`
+	RevertReason string          `json:"revertReason,omitempty"`
+	Logs         []callLog       `json:"logs,omitempty" rlp:"optional"`
+	// Placed at end on purpose. The RLP will be decoded to 0 instead of
+	// nil if there are non-empty elements after in the struct.
+	Value            *big.Int `json:"value,omitempty" rlp:"optional"`
+	revertedSnapshot bool
+
+	AccessedSlots     accessedSlots                              `json:"accessedSlots"`
+	ExtCodeAccessInfo []common.Address                           `json:"extCodeAccessInfo"`
+	UsedOpcodes       map[vm.OpCode]uint64                       `json:"usedOpcodes"`
+	ContractSize      map[common.Address]*contractSizeWithOpcode `json:"contractSize"`
+	OutOfGas          bool                                       `json:"outOfGas"`
+	Calls             []callFrameWithOpcodes                     `json:"calls,omitempty" rlp:"optional"`
+}
+
+func (f callFrameWithOpcodes) TypeString() string {
+	return f.Type.String()
+}
+
+func (f callFrameWithOpcodes) failed() bool {
+	return len(f.Error) > 0 && f.revertedSnapshot
+}
+
+func (f *callFrameWithOpcodes) processOutput(output []byte, err error, reverted bool) {
+	output = common.CopyBytes(output)
+	// Clear error if tx wasn't reverted. This happened
+	// for pre-homestead contract storage OOG.
+	if err != nil && !reverted {
+		err = nil
+	}
+	if err == nil {
+		f.Output = output
+		return
+	}
+	f.Error = err.Error()
+	f.revertedSnapshot = reverted
+	if f.Type == vm.CREATE || f.Type == vm.CREATE2 {
+		f.To = nil
+	}
+	if !errors.Is(err, vm.ErrExecutionReverted) || len(output) == 0 {
+		return
+	}
+	f.Output = output
+	if len(output) < 4 {
+		return
+	}
+	if unpacked, err := abi.UnpackRevert(output); err == nil {
+		f.RevertReason = unpacked
+	}
+}
+
+type callFrameWithOpcodesMarshaling struct {
+	TypeString string `json:"type"`
+	Gas        hexutil.Uint64
+	GasUsed    hexutil.Uint64
+	Value      *hexutil.Big
+	Input      hexutil.Bytes
+	Output     hexutil.Bytes
+}
+
+type accessedSlots struct {
+	Reads           map[string][]string `json:"reads"`
+	Writes          map[string]uint64   `json:"writes"`
+	TransientReads  map[string]uint64   `json:"transientReads"`
+	TransientWrites map[string]uint64   `json:"transientWrites"`
+}
+
+type opcodeWithPartialStack struct {
+	Opcode        vm.OpCode
+	StackTopItems []uint256.Int
+}
+
+type erc7562Tracer struct {
+	config    erc7562TracerConfig
+	gasLimit  uint64
+	depth     int
+	interrupt atomic.Bool // Atomic flag to signal execution interruption
+	reason    error       // Textual reason for the interruption
+	env       *tracing.VMContext
+
+	ignoredOpcodes       map[vm.OpCode]struct{}
+	callstackWithOpcodes []callFrameWithOpcodes
+	lastOpWithStack      *opcodeWithPartialStack
+	Keccak               map[string]struct{} `json:"keccak"`
+}
+
+// catchPanic handles panic recovery and logs the panic and stack trace.
+func catchPanic() {
```

Comment:
I really don't like this pattern. Your code should be correct, you should not rely on recovering. Rather fail fast and fix quickly

Reply:
Right, I'd like to second this. Did you need this for debug purposes?

Panics here should not crash the node. API internal has recovery mechanism from panics. If you managed to crash the node then that's an issue we should look at.

Reply:
I needed that for debugging purposes, since the errors on the hooks were silently ignored by geth it seems. I'm removing it.